azerothcore/azerothcore-wotlk
# fix(DB/Creature): Emeraldon Mobs Missing Patrol Patterns

## Changes Proposed:
-  Add formations for the 7 packs of Emeraldon mobs and set them to wander.
-  Took the wander distance from other similar dragonkin npcs from the other world boss areas.

## Issues Addressed:

- Closes #6218
- Closes chromiecraft/chromiecraft#771

## SOURCE:


## Tests Performed:

- See next section.

## How to Test the Changes:


DB TEST
1. Run this query to observe the changes:
```
SET @GROUP1_LEADER = 33210;
SET @GROUP1_LEFT = 33226;
SET @GROUP1_RIGHT = 33241;

SET @GROUP2_LEADER = 33243;
SET @GROUP2_LEFT = 33229;
SET @GROUP2_RIGHT = 33228;

SET @GROUP3_LEADER = 33240;
SET @GROUP3_LEFT = 33209;
SET @GROUP3_RIGHT = 33208;

SET @GROUP4_LEADER = 33242;
SET @GROUP4_LEFT = 33211;
SET @GROUP4_RIGHT = 33227;

SET @GROUP5_LEADER = 33245;
SET @GROUP5_LEFT = 33230;
SET @GROUP5_RIGHT = 33231;

SET @GROUP6_LEADER = 33246;
SET @GROUP6_LEFT = 33214;
SET @GROUP6_RIGHT = 33215;

SET @GROUP7_LEADER = 33244;
SET @GROUP7_LEFT = 33212;
SET @GROUP7_RIGHT = 33213;
SELECT * FROM `creature` WHERE `guid` IN (@GROUP1_LEADER, @GROUP2_LEADER, @GROUP3_LEADER, @GROUP4_LEADER, @GROUP5_LEADER, @GROUP6_LEADER, @GROUP7_LEADER);
SELECT * FROM `creature_formations` WHERE `memberGUID` IN (
@GROUP1_LEADER, @GROUP2_LEADER, @GROUP3_LEADER, @GROUP4_LEADER, @GROUP5_LEADER, @GROUP6_LEADER, @GROUP7_LEADER,
@GROUP1_LEFT, @GROUP2_LEFT, @GROUP3_LEFT, @GROUP4_LEFT, @GROUP5_LEFT, @GROUP6_LEFT, @GROUP7_LEFT,
@GROUP1_RIGHT, @GROUP2_RIGHT, @GROUP3_RIGHT, @GROUP4_RIGHT, @GROUP5_RIGHT, @GROUP6_RIGHT, @GROUP7_RIGHT
);
```
2. The first query returns the results of the 7 formation leaders. Their MovementType should be set to 1, and their wander_distance should be set to 15.
3. The second query returns the npc formations. There should be 7 groups of 3 grouping by the leaderGUID. 
Each group should have 1 entry where the leaderGUID and memberGUID is the same (this is the leader). All the other columns for the leader should be 0.
The other 2 entries in the group should have a groupAI set to 2. One entry will have an angle of 300 and the other will have an angle of 60

IN-GAME TEST
1. Move to one of the npcs in the area (ex. ```go creature 33210```)
2. Observe the npcs in the area. There should be 7 groups of 3 npcs each. Each group should wander together, staying together as a 3-npc unit rather than wandering separatly.

## Known Issues and TODO List:


- [Other world dragon npcs need to be looked at. Can guarantee that at least Feralas npcs need to be fixed.]


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
